File: parrot_spam.hs
import Control.Monad
import Data.List
import System.Environment
import System.Exit
import System.IO

main :: IO ()
main = do
    parrots <- filter (not . null) <$> getArgs
    when (null parrots) $ die "Nothing to repeat"

    isatty <- hIsTerminalDevice stdout
    (if isatty then putStrLn else putStr) $ spam (cycle parrots) 4000

spam :: [String] -> Int -> String
spam (p : ps) avail =
    let avail' = avail - length p
    in if avail' >= 0 then p ++ spam ps avail' else ""

File: parrot_spam.jl
function spam(chars, parrots)
    buffer = IOBuffer(maxsize = chars)

    for parrot in Iterators.cycle(parrots)
        chars -= length(parrot)
        if chars < 0
            break
        end

        write(buffer, parrot)
    end

    return String(take!(buffer))
end

parrots = filter(!isempty, ARGS)
if isempty(parrots)
    println("Nothing to repeat")
    exit(1)
end

println(spam(4000, parrots))
